app/views/feedback.py

```diff
@@ -0,0 +1,93 @@
+from flask import Blueprint, redirect, url_for, escape, request, current_app, g
+from flask_login import login_required, current_user
+from flask_wtf import FlaskForm
+from wtforms import StringField, TextAreaField
+from wtforms.validators import InputRequired
+
+from app.helpers.form_helper import is_valid_form
+from app.helpers import template_helper
+from app.submitter.encrypter import encrypt
+from app.submitter.submission_failed import SubmissionFailedException
+from app.globals import get_metadata
+from app.submitter.converter import convert_feedback
+from app.utilities.schema import load_schema_from_metadata
+
+from structlog import get_logger
+
+logger = get_logger()
+
+feedback_blueprint = Blueprint(name='feedback',
+                               import_name=__name__,
+                               url_prefix='/feedback')
+
+
+class FeedbackForm(FlaskForm):
+    message = TextAreaField('message', validators=[InputRequired()])
+    name = StringField('name')
+    email = StringField('email')
+
+
+@feedback_blueprint.before_request
+def before_request():
+    metadata = get_metadata(current_user)
+    if metadata:
+        logger.bind(tx_id=metadata['tx_id'])
+        g.schema_json = load_schema_from_metadata(metadata)
+
+
+@feedback_blueprint.route('', methods=['GET'])
+@login_required
+def get_form():
+    form = FeedbackForm()
+    return _render_template(
+        'feedback',
+        page_title='Survey feedback',
+        form=form,
+    )
+
+
+@feedback_blueprint.route('', methods=['POST'])
+@login_required
+def send_feedback():
+    form = FeedbackForm()
+
+    if is_valid_form(form):
+
+        metadata = get_metadata(current_user)
+
+        message = convert_feedback(
+            metadata,
+            {
+                'message': escape(request.form.get('message')),
+                'name': escape(request.form.get('name')),
+                'email': escape(request.form.get('email')),
+                'referer': request.referrer or '',
```

Comment:
- I don't think the property should be called `referer`, that's just how we're getting the URL that they are providing feedback for. Something more generic like `url` would be better.
- I think the four message properties should be passed explicitly to the convert_feedback method - `convert_feedback(message, name, email, url, metadata, survey_id)`, as it stands it implies any payload can be provided